kits/daisy/src/main.cpp

```diff
@@ -0,0 +1,648 @@
+  }
+  else if (is_partial)
+  {
+    std::cout << "Creating partial hexapod for testing" << std::endl;
+    hexapod = Hexapod::createPartial(params, legs, hex_errors);
+  }
+  else
+  {
+    std::cout << "Searching network for modules..." << std::endl;
+    hexapod = Hexapod::create(params, hex_errors);
+  }
+
+  if (!hexapod)
+  {
+    if (!is_quiet)
+    {
+      QMessageBox param_error;
+      param_error.setWindowTitle("Could not start up hexapod");
+      param_error.setText("Could not find hexapod on network -- check that all modules are visible via the Scope GUI.");
+      param_error.exec();
+    }
+    return nullptr;
+  }
+
+  // Clear any colors potentially set elsewhere
+  hexapod->clearLegColors();
+
+  if(!hex_errors.has_valid_initial_feedback)
+  {
+    if (!is_quiet)
+    {
+      QMessageBox param_error;
+      param_error.setWindowTitle("Did not get initial feedback from modules!");
+      param_error.setText("Could not get feedback from all modules -- this could indicate an intermittent network connection with the modules.");
+      param_error.exec();
+    }
+    return nullptr;
+  }
+
+  if(hex_errors.first_out_of_range_leg != -1)
+  {
+    if (!is_quiet)
+    {
+      QMessageBox param_error;
+      param_error.setWindowTitle("Base module out of range!");
+      std::string message = "Base module out of range (leg " + std::to_string(hex_errors.first_out_of_range_leg + 1) + ")";
+      param_error.setText(QString(message.c_str()));
+      param_error.exec();
+    }
+    else
+    {
+      // Set to red!
+      hexapod->setLegColor(hex_errors.first_out_of_range_leg, 255, 0, 0);
+    }
+    return nullptr;
+  }
+
+  if (hex_errors.m_stop_pressed)
+  {
+    if (!is_quiet)
+    {
+      QMessageBox param_error;
+      param_error.setWindowTitle("M-stop active!");
+      param_error.setText("M-stop is active on at least one module! Reset before continuing.");
+      param_error.exec();
+    }
+    return nullptr;
+  }
+
+  return hexapod;
+
+}
+
+int main(int argc, char** argv)
+{
+  // Construct initially so I can pop up error dialogs!
+  QApplication app(argc, argv);
+
+  bool do_visualize{};
+  bool is_dummy{};
+  bool is_partial{};
+  bool is_quiet{};
+  std::set<int> legs;
+  if (!parse_parameters(argc, argv, do_visualize, is_dummy, is_partial, is_quiet, legs))
+    return -1;
+
+  HexapodParameters params;
+  if (!params.loadFromFile("hex_config.xml"))
+  {
+    params.resetToDefaults();
+    if (!is_quiet)
+    {
+      QMessageBox param_error;
+      param_error.setWindowTitle("Could not find parameters");
+      param_error.setText("Could not find hexapod parameters, or file corrupt! Using default parameters.");
+      param_error.exec();
+    }
+  }
+
+  std::unique_ptr<Hexapod> hexapod = getHexapod(params, is_dummy, is_partial, is_quiet, legs);
+  while (is_quiet && !hexapod)
+  {
+    hexapod = getHexapod(params, is_dummy, is_partial, is_quiet, legs);
+  }
+
+  if(!hexapod && !is_quiet)
+    return -1;
```

Comment:
Return codes are practically never negative, so change to `+1`.

See [this](https://unix.stackexchange.com/questions/418784/what-is-the-min-and-max-values-of-exit-codes-in-linux) for more